### ISSUE TITLE
One question about test.py

### ISSUE THREAD

**Issue by kunsung** [xingyizhou/CenterNet#574]
Why do tests use val data instead of test data?
![2020-01-06 21-20-01屏幕截图](https://user-images.githubusercontent.com/38248825/71820844-689f8300-30cb-11ea-99e9-28ed95505fd7.png)

**Comment by xingyizhou** [comment]
We don't have annotations for the test set.

**Comment by kunsung** [comment]
> We don't have annotations for the test set.

So can a validation set be used for both validation and testing? Or use train sets for training and validation, and validation sets for testing.
![1](https://user-images.githubusercontent.com/38248825/72036513-0797cb00-32d6-11ea-93a9-e3d69f7e7240.png)

**Comment by xingyizhou** [comment]
We used the validation set for validation and reported validation accuracy. We also reported the testing set accuracy for testing in the paper. Remark that the testing set should be only used once for each method, so most of the experiments are done on the validation set.

**Comment by jhonjam** [comment]
@xingyizhou  Why the result.json file generated from the test.py execution contains for each associated image_id different category_id. This can be seen in the following figure.
![result](https://user-images.githubusercontent.com/62752094/91997694-c2b31f80-ed10-11ea-8aef-d5ccf3748df4.PNG)

For a better observation, I organize for each "image_id" the values corresponding to "category_id", "score" and "bbox" in a .txt file. As can be seen, there are too many "category_id" for each of the classes, which is an error, since in the image that I am using there are no more than 8 objects to detect. I would like to know, how can I get the true corresponding values "category_id", "score" and "bbox" for each real object in each image?

Tank you

`0 0.24 984.53 182.43 58.82 67.02 
0 0.23 882.65 59.76 46.18 49.04 
0 0.06 566.62 279.26 137.62 111.52 
0 0.03 1097.51 -304.42 352.98 54.89 
0 0.03 459.99 190.57 327.99 431.83 
0 0.03 287.82 197.02 513.99 338.87 
0 0.03 923.98 -296.0 240.65 34.37 
0 0.02 7.76 -281.95 -6.55 10.59 
0 0.02 258.88 199.57 532.62 375.38 
0 0.02 263.23 221.64 504.71 371.58 
0 0.02 -26.33 -292.65 261.51 47.78 
0 0.02 693.39 288.94 58.55 112.11 
0 0.02 -177.7 -371.97 645.09 201.33 
0 0.02 0.3 688.98 2.78 49.31 
0 0.02 1253.79 39.47 46.75 16.96 
0 0.02 677.39 350.27 75.49 86.89 
0 0.02 37.5 -332.97 375.58 132.04 
0 0.02 73.45 -329.87 343.7 128.41 
0 0.02 78.15 -330.35 373.54 125.64 
0 0.02 226.86 -316.05 194.17 90.83 
0 0.02 945.58 78.58 25.85 33.14 
0 0.02 1275.63 -64.5 4.06 58.97 
1 0.63 882.65 59.76 46.18 49.04 
1 0.2 300.48 223.73 491.07 381.88 
1 0.1 984.53 182.43 58.82 67.02 
1 0.05 11.35 -6.64 61.26 43.75 
1 0.03 520.73 -8.25 45.12 25.7 
1 0.03 551.16 227.95 231.58 160.63 
1 0.03 566.62 279.26 137.62 111.52 
1 0.03 282.72 587.22 284.96 135.77 
1 0.03 1237.48 1.18 36.07 60.61 
1 0.03 1255.62 13.39 43.52 44.32 
1 0.02 31.05 127.85 723.22 574.08 
1 0.02 637.55 231.89 117.12 43.82 
1 0.02 -1.94 0.2 5.65 48.23 
1 0.02 286.08 579.99 476.51 129.41 
1 0.02 0.19 311.96 0.58 27.93 
2 0.3 300.48 223.73 491.07 381.88 
2 0.13 882.65 59.76 46.18 49.04 
2 0.09 984.53 182.43 58.82 67.02 
2 0.08 551.52 230.3 230.55 167.71 
2 0.05 458.74 187.72 330.33 453.14 
2 0.03 31.05 127.85 723.22 574.08 
2 0.03 677.39 350.27 75.49 86.89 
2 0.03 1255.62 13.39 43.52 44.32 
2 0.03 -14.89 -0.35 397.99 431.35 
2 0.02 693.39 288.94 58.55 112.11 
2 0.02 326.55 -140.05 77.32 98.19 
2 0.02 -10.65 10.64 388.8 268.71 
2 0.02 1097.51 -304.42 352.98 54.89 
2 0.02 1011.44 76.68 84.8 160.19 
2 0.02 520.73 -8.25 45.12 25.7 
2 0.02 1237.48 1.18 36.07 60.61 
2 0.02 7.76 -281.95 -6.55 10.59 
2 0.02 -1.94 0.2 5.65 48.23 
2 0.02 94.04 66.45 283.36 97.48 
2 0.02 996.51 20.93 176.41 207.94 
2 0.02 1276.53 -48.9 2.68 47.23 
2 0.02 0.19 311.96 0.58 27.93 
3 0.1 882.65 59.76 46.18 49.04 
3 0.1 300.48 223.73 491.07 381.88 
3 0.05 984.53 182.43 58.82 67.02 
3 0.03 1255.62 13.39 43.52 44.32 
3 0.03 551.52 230.3 230.55 167.71 
3 0.03 -130.83 959.01 272.0 74.67 
3 0.02 677.39 350.27 75.49 86.89 
3 0.02 1045.63 951.01 456.62 80.68 
3 0.02 865.25 -297.53 298.82 36.62 
3 0.02 73.54 872.81 62.38 109.3 
3 0.02 -144.31 925.2 376.86 133.4 
3 0.02 1198.29 75.37 159.08 298.92 
3 0.02 1189.12 72.96 177.48 263.64 
3 0.02 1178.83 66.81 197.7 236.0 
3 0.02 1276.53 -48.9 2.68 47.23 
3 0.02 1183.97 688.87 177.74 91.94 
4 0.1 883.82 53.24 55.93 61.59 
4 0.09 300.48 223.73 491.07 381.88 
4 0.04 551.52 230.3 230.55 167.71 
4 0.03 458.74 187.72 330.33 453.14 
4 0.03 556.45 268.58 196.58 212.48 
4 0.03 637.55 231.89 117.12 43.82 
4 0.03 984.53 182.43 58.82 67.02 
4 0.02 1045.63 951.01 456.62 80.68 
4 0.02 -130.83 959.01 272.0 74.67 
4 0.02 109.1 897.82 32.0 59.67 
4 0.02 284.6 677.83 279.53 78.62 
4 0.02 -88.64 888.69 528.05 201.53 
4 0.02 257.67 667.96 553.41 135.59 
5 0.04 244.34 228.54 523.4 372.83 
5 0.04 567.88 279.21 151.27 111.25 
5 0.04 551.52 230.3 230.55 167.71 
5 0.04 309.65 231.15 485.74 383.3 
5 0.04 984.53 182.43 58.82 67.02 
5 0.04 281.71 186.92 527.01 533.75 
5 0.04 287.82 197.02 513.99 338.87 
5 0.04 368.77 206.45 449.14 414.4 
5 0.03 882.65 59.76 46.18 49.04 
5 0.02 693.39 288.94 58.55 112.11 
5 0.02 -177.7 -371.97 645.09 201.33 
5 0.02 1255.62 13.39 43.52 44.32 `

**Comment by chordxu** [comment]
Could you tell me how to change the code to use test data ?
